src/App.js

```diff
@@ -1,16 +1,32 @@
 import React from 'react';
 import './App.css';
+import { useState } from 'react';
+import ChatEntry from './components/ChatEntry';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
 
 const App = () => {
+  
+  const [messageData,setMessageData] = useState(chatMessages)
+  
+  const likeMessage = (id) => {
+    setMessageData = messageData => messageData.map(message => {
```

Comment:
Close! I think you meant to call the update function `setMessageData` instead of trying to reassign it, like so:
```javascript
    setMessageData((messageData) =>
      messageData.map((message) => {
        if (message.id === id) {
          return {
            ...message,
            liked: !message.liked,
          };
        } else {
          return message;
        }
      })
    );
```
Note the callback function passed to `setMessageData` and that you should set `liked` by flipping the previous `liked` state.